### PR TITLE
[phpstan-extensions] Provide phpstan container to bootstrap file from extensions

### PR DESCRIPTION
**Pull request by TomasVotruba**
This might help with: https://github.com/rectorphp/rector/issues/7795

### CODE REVIEW

File: src/Autoloading/BootstrapFilesIncluder.php [hunk 1]
```diff
@@ -87,10 +88,16 @@ private function resolveAbsoluteBootstrapFilePaths(array $extensionConfigFiles):
         return $absoluteBootstrapFilePaths;
     }
 
-    private function tryRequireFile(string $bootstrapFile): void
+    /**
+     * PHPStan container mimics:
+     * https://github.com/phpstan/phpstan-src/blob/34881e682e36e30917dcfa8dc69c70e857143436/src/Command/CommandHelper.php#L513-L515
+     */
+    private function tryRequireFile(string $bootstrapFile, ?Container $phpstanContainer = null): void
```
Comment (samsonasik) [inline src/Autoloading/BootstrapFilesIncluder.php]: I think variable should be `$container`, same with what phpstan does, as it used via `use()` in closure to be used inside require_once, not as parameter in closure.

Comment (TomasVotruba) [reply src/Autoloading/BootstrapFilesIncluder.php]: Could you add the diff snippet of what you mean?

Comment (samsonasik) [reply src/Autoloading/BootstrapFilesIncluder.php]: Yes:

```suggestion
    private function tryRequireFile(string $bootstrapFile, ?Container $container = null): void
```

File: src/Autoloading/BootstrapFilesIncluder.php [hunk 2]
```diff
@@ -87,10 +88,16 @@ private function resolveAbsoluteBootstrapFilePaths(array $extensionConfigFiles):
...
     {
         try {
-            require_once $bootstrapFile;
+            (static function (string $bootstrapFile) use ($phpstanContainer): void {
```
Comment (samsonasik) [inline src/Autoloading/BootstrapFilesIncluder.php]:
```suggestion
            (static function (string $bootstrapFile) use ($container): void {
```

File: phpstan.neon [hunk 3]
```diff
@@ -812,3 +812,11 @@ parameters:
         - '#Call to deprecated method getDirectClassNames\(\) of class PHPStan\\Type\\TypeUtils.*#'
         - '#Parameter 3 should use "PHPStan\\Reflection\\ParameterReflectionWithPhpDocs" type as the only type passed to this method#'
         - '#Parameters should use "PHPStan\\Reflection\\ParameterReflectionWithPhpDocs\|PhpParser\\Node\\Param" types as the only types passed to this method#'
+
+        # various usage
+        - '#Parameters should use "PHPStan\\DependencyInjection\\Container" types as the only types passed to this method#'
+
+        # actually used in global scope
+        -
+            message: '#Anonymous function has an unused use \$phpstanContainer#'
```
Comment (samsonasik) [inline phpstan.neon]:
```suggestion
            message: '#Anonymous function has an unused use \$container#'
```

**Review by samsonasik — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thanks for checking @samsonasik :+1:

**Pull request merged**